### DISCUSSION TITLE
How to find and scroll to the result without replacing?

### DISCUSSION THREAD

**Discussion by vvkdev** [ArturoDent/find-and-transform#11] [Q&A] [answered]
Hi, I'm trying to use a keybinding to find and select regex matching, and it works, but without scrolling to the new cursor position.
Please, help me to improve my command:
```
"key": "alt+q",
   "command": "findInCurrentFile",
   "args": {
   "find": "<tag(.|\n)*?</tag>",
   "isRegex": true,
      }
```
I tried to add "restrictFind": "nextMoveCursor" or "restrictFind": "nextSelect", but then it doesn't select anything at all.

Thank you!

**Comment by ArturoDent** [comment]
Hi vvkdev.  I am looking at this.  One question: should the first match in the file be revealed - even if it is above the cursor - or should the next match after the cursor be revealed?

**Reply by vvkdev** [reply]
Hi! Good question. I think, there is should be an option, but if not, the first match would be more expected. It's important to place the first cursor to the first match, because only in this case the Enumerate extension, as example, and the like will work in the correct order.
So it would be logical to scroll to the first match too, imho.
And is there an option to select all matches or the first/next one only?

**Comment by ArturoDent** [comment, marked as answer]
I added a `reveal` option to the arguments.  Install v4.4.5 of the extension.  

`reveal` options are `first`, `next`, and `last`. What they do exactly is influenced by whether you are searching in the entire docuemnt (as you are) or in a line or selection.

In your case this keybinding should work:

```jsonc
{   
  "key": "alt+q",
  "command": "findInCurrentFile",
  "args": {
    "find": "<tag(.|\n)*?</tag>",
    "isRegex": true,
    "reveal": "next"           // this is new
  }
}
```

Let me know if it works for you.

**Reply by vvkdev** [reply]
I've tested this new option a little bit and it seems to work fine ))
Thank you!!
